full/src/test/java/apoc/custom/CypherProceduresTest.java

```diff
@@ -240,6 +240,121 @@ public void shouldRemoveTheCustomProcedure() throws Exception {
+                "Please ensure you've spelled the procedure name correctly and that the procedure is properly deployed.");
+        thrown.expect(new StatusCodeMatcher("Neo.ClientError.Statement.SyntaxError"));
+
+        // given
+        db.executeTransactionally("call apoc.custom.asProcedure('a.b.c','RETURN 42 as answer')");
+        TestUtil.testCall(db, "call custom.a.b.c()", (row) -> assertEquals(42L, ((Map)row.get("row")).get("answer")));
+        db.executeTransactionally("call db.clearQueryCaches()");
+        db.executeTransactionally("call apoc.custom.asProcedure('a.b.c','RETURN 43 as answer')");
+        TestUtil.testCall(db, "call custom.a.b.c()", (row) -> assertEquals(43L, ((Map)row.get("row")).get("answer")));
+        db.executeTransactionally("call db.clearQueryCaches()");
+
+        // when
+        db.executeTransactionally("call apoc.custom.removeProcedure('a.b.c')");
+        db.executeTransactionally("call db.clearQueryCaches()");
+
+        // then
+        TestUtil.count(db, "call custom.a.b.c()");
+    }
+
+    @Test
+    public void shouldRemoveTheCustomFunctionWithDotInName() throws Exception {
+        thrown.expect(QueryExecutionException.class);
+        thrown.expectMessage("Unknown function 'custom.a.b.c'");
+        thrown.expect(new StatusCodeMatcher("Neo.ClientError.Statement.SyntaxError"));
+
+        // given
+        db.executeTransactionally("CALL apoc.custom.asFunction('a.b.c','RETURN 42 as answer')");
+        TestUtil.testCall(db, "return custom.a.b.c() as row", (row) -> assertEquals(42L, ((Map)((List)row.get("row")).get(0)).get("answer")));
+
+        // when
+        db.executeTransactionally("call apoc.custom.removeFunction('a.b.c')");
+        db.executeTransactionally("call db.clearQueryCaches()");
+
+        // then
+        TestUtil.count(db, "RETURN custom.a.b.c()");
+    }
+
+    @Test
+    public void shouldRemoveTheCustomProcedureWithDotInName() throws Exception {
+        thrown.expect(QueryExecutionException.class);
+        thrown.expectMessage("There is no procedure with the name `custom.a.b.c` registered for this database instance. " +
+                "Please ensure you've spelled the procedure name correctly and that the procedure is properly deployed.");
+        thrown.expect(new StatusCodeMatcher("Neo.ClientError.Statement.SyntaxError"));
+
+        // given
+        db.executeTransactionally("call apoc.custom.asProcedure('a.b.c','RETURN 42 as answer')");
+        TestUtil.testCall(db, "call custom.a.b.c()", (row) -> assertEquals(42L, ((Map)row.get("row")).get("answer")));
+
+        // when
+        db.executeTransactionally("call apoc.custom.removeProcedure('a.b.c')");
+        db.executeTransactionally("call db.clearQueryCaches()");
+
+        // then
+        TestUtil.count(db, "call custom.a.b.c()");
+    }
+
+    @Test
+    public void shouldOverrideCustomFunctionWithDotInNameOnlyIfWithSameNamespaceAndFinalName() throws Exception {
```

Comment:
might be worth testing list and removal of such functions (e.g.: list displays both, removing 1 does not remove the other...)

Reply:
Well yes, in this way I found another case that I had not considered :( . 
Now I have updated the code and the test case.